packages/rspack/src/executors/rspack/rspack.impl.ts

```diff
@@ -111,3 +118,64 @@ function registerCleanupCallback(callback: () => void) {
   process.on('SIGTERM', wrapped);
   process.on('exit', wrapped);
 }
+
+async function runTypeCheck(
```

Comment:
Let's use the util `runTypeCheck` and `printDiagnostic` from `@nx/js`.

See an example here: https://github.com/nrwl/nx/blob/master/packages/vite/src/utils/executor-utils.ts#L11-L27